src/util/system.cpp

```diff
@@ -741,6 +750,10 @@ const fs::path &GetBlocksDir()
 
 const fs::path &GetDataDir(bool fNetSpecific)
 {
+#ifndef WIN32
```

Comment:
GetDataDir, a getter, is not the place to do things such as set the umask. Please don't add these kind of side-effects.

Reply:
Maybe `GetDataDir` is not a proper name for this function as it already has an intended side-effect:
https://github.com/bitcoin/bitcoin/blob/ddd3cd30cb050fe9b8738778beecba1d8bd50d52/src/util/system.cpp#L777-L780

Reply:
I know. I kind of want to get rid of that too. But don't add any more.

Reply:
Done.